private/pkg/storage/bucket.go

```diff
@@ -52,6 +52,38 @@ type ReadBucket interface {
 	Walk(ctx context.Context, prefix string, f func(ObjectInfo) error) error
 }
 
+// PutOptions are the possible options that can be passed to a Put operation.
+type PutOptions struct {
+	IfNotExists      bool
+	CustomChunkSize  bool
+	ChunkSizeInBytes int
+}
+
+// PutOption are options passed when putting an object in a bucket.
+type PutOption func(*PutOptions)
```

Comment:
All other options structs in `storage` are private and at the end of the file, see `diff.go` as an example.

Reply:
Right, but all implementations need to read and understand the passed options outside of this pkg, so probably exported struct with private fields and exported getters?

Reply:
Just to follow up on this - talked offline, @unmultimedio is right